exercises/all-your-base/src/example/kotlin/BaseConverter.kt

```diff
@@ -0,0 +1,76 @@
+class BaseConverter(originalBase: Int, originalDigits: IntArray) {
+
+    companion object {
+        private val MINIMUM_VALID_BASE = 2
+        private val INVALID_BASE_ERROR_MESSAGE = "Bases must be at least 2."
+    }
+
+    private val numeral: Int
+
+    init {
+        validateInputs(originalBase, originalDigits)
+        this.numeral = computeNumeral(originalBase, originalDigits)
+    }
+
+    fun convertToBase(newBase: Int): IntArray {
+        if (newBase < MINIMUM_VALID_BASE) throw IllegalArgumentException(INVALID_BASE_ERROR_MESSAGE)
+
+        val largestExponent = computeLargestExponentForBase(newBase)
+        val result = IntArray(largestExponent + 1)
+        var remainder = numeral
+
+        for (currentExponent in largestExponent downTo 0) {
+            val coefficient = Math.floor(remainder / Math.pow(newBase.toDouble(), currentExponent.toDouble())).toInt()
+
+            result[largestExponent - currentExponent] = coefficient
+
+            remainder -= (coefficient * Math.pow(newBase.toDouble(), currentExponent.toDouble())).toInt()
+        }
+
+        return result
+    }
+
+    private fun validateInputs(originalBase: Int, originalDigits: IntArray) {
+        if (originalBase < MINIMUM_VALID_BASE) {
```

Comment:
All of these if/throw statements can be simplified by using Kotlin's `require` method, see an example here: https://github.com/exercism/xkotlin/blob/02ecc5a70719f0b0e741992f33879f880f7c407f/exercises/pascals-triangle/src/example/kotlin/PascalsTriangle.kt#L4